Source/Controllers/Other/SQLitePinnedTableManager.swift

```diff
@@ -0,0 +1,207 @@
+                    Log.debug("startingSchemaVersion = \(startingSchemaVersion)")
+                }
+                rs.close()
+
+                schemaBlock(db, Int(startingSchemaVersion))
+
+                if newSchemaVersion != startingSchemaVersion, newSchemaVersion > 0 {
+                    let query = "PRAGMA user_version = " + String(newSchemaVersion)
+                    Log.debug("query = \(query)")
+                    try db.executeUpdate(query, values: nil)
+                } else {
+                    Log.info("db schema did not need an update")
+                }
+            } catch {
+                Log.error("Something went wrong: \(error.localizedDescription)")
+            }
+        }
+    }
+
+    private func loadPinnedTablesHistory() {
+
+        queue.inDatabase { db in
+            do {
+                db.traceExecution = traceExecution
+                // select by id desc to get latest first
+                let rs = try db.executeQuery("SELECT hostName, databaseName, pinnedTableName FROM PinnedTables order by id desc", values: nil)
+
+                while rs.next() {
+                    let hostName = rs.string(forColumn: "hostname")!
+                    let databaseName = rs.string(forColumn: "databaseName")!
+                    let pinnedTableName = rs.string(forColumn: "pinnedTableName")!
+                    addToPinnedTablesDatabaseDictionary(hostName: hostName, databaseName: databaseName, tableToPin: pinnedTableName)
+                }
+                rs.close()
+            } catch {
+                logDBError(error)
+            }
+        }
+        queue.close()
+    }
+
+    @objc func getPinnedTables(hostName: String, databaseName: String) -> [String] {
+        return pinnedTablesDatabaseDictionary[hostName]?[databaseName] ?? []
+    }
+
+    @objc func pinTable(hostName: String, databaseName: String, tableToPin: String) {
+        
+        if (pinnedTablesDatabaseDictionary[hostName]?[databaseName] != nil && pinnedTablesDatabaseDictionary[hostName]![databaseName]!.contains(tableToPin)) {
+            return // tableToPin already pinned
+        }
+        addToPinnedTablesDatabaseDictionary(hostName: hostName, databaseName: databaseName, tableToPin: tableToPin)
+        queue.inDatabase { db in
+            db.traceExecution = traceExecution
+            do {
+                try db.executeUpdate("INSERT INTO PinnedTables (hostName, databaseName, pinnedTableName) VALUES (?, ?, ?)",
+                        values: [hostName, databaseName, tableToPin])
+            } catch {
+                logDBError(error)
+            }
+        }
+        queue.close()
+    }
+    
+
+    @objc func unpinTable(hostName: String, databaseName: String, tableToUnpin: String) {
+        if pinnedTablesDatabaseDictionary[hostName]?[databaseName] == nil {
+            return
+        }
+        if !pinnedTablesDatabaseDictionary[hostName]![databaseName]!.contains(tableToUnpin) {
```

Comment:
^^ - try to avoid force unwrap via getting the `database` to `let` variable instead of check for null

```swift
guard let database = pinnedTablesDatabaseDictionary[hostName]?[databaseName], database.contains(tableToUnpin) else {
   return
}
```